spec/unit/transaction/additional_resource_generator_spec.rb

```diff
@@ -182,6 +182,31 @@ def find_vertex(graph, type, title)
         'Generator[thing]', 'Whit[completed_thing]')
     end
 
+    it "should tag the sentinel with the tags of the resource" do
+      graph = relationships_after_eval_generating(<<-MANIFEST, 'Generator[thing]')
+        generator { thing:
+          code => 'notify { hello: }',
+          tag  => 'foo',
+        }
+      MANIFEST
+      whit = find_vertex(graph, :whit, "completed_thing")
+      # this matcher relies on Puppet::Util::Tagging#tagged? which accepts a
+      # list of tags and returns true if _at least one_ tag is present. So we
+      # explicitly check both.
+      expect(whit).to be_tagged("thing")
+      expect(whit).to be_tagged("foo")
+    end
+
+    it "should not tag the sentinel with the type of the resource" do
+      graph = relationships_after_eval_generating(<<-MANIFEST, 'Generator[thing]')
+        generator { thing:
+          code => 'notify { hello: }',
+        }
+      MANIFEST
+      whit = find_vertex(graph, :whit, "completed_thing")
+      expect(whit).not_to be_tagged("generator")
+    end
+
```

Comment:
@Iristyle got something rigged up that should be what we need here. Per conversation, removed `tags.delete` of generating resource type name, and added a new test to the integration section that should cover the use case of triggering dependent resources from dynamically `eval_generated` resources using implicit tagging. Since as far as I can tell none of the existing puppet types support both dynamically generating resources with `eval_generate` and also something like a `refreshonly` parameter out of the box, we temporarily augment the exec type with `eval_generate` to replicate what we would need for the use case.

Word to the wise - don't stub `eval_generate` with a single return value (even one that's dynamic) - no matter how many times its called, it will always return the first incantation, leading to an infinite explosion in graph edges during transaction evaluation as new edges "mysteriously" multiply to the exact same generated resource instance.

I.e., an initial iteration of this did:
```ruby
Puppet::Type::Exec.any_instance.stubs(:eval_generate).returns (
  [ (Puppet::Type.type(:notify).new :name => "notify#{rand(9999)}") ]
)
```

Stubbing instead with a list of comma-separated return values will return the first value on first incantation of the stubbed method, second value with the second incantation, and so on.